project/Game.py

```diff
@@ -149,8 +149,43 @@ def start(self):
     def end(self):
         self.ended = True
 
-    def quit_question(self, team, password):
-        return True
+    def quit_question(self, dtn , team, password):
+        if(team.password==password):
+            team.currentLandmark+=1
+            if(dtn<team.clueTime):
+                team.timelog(datetime.timedelta(days=0,hours=0,minutes=0,seconds=0))
+            else:
+                team.timelog.append(dtn-team.clueTime)
+            team.clueTime=dtn
+            team.penaltyCount = 0
+            return True
+        return False
+
+    def answer_question(self, dtn, team, answer):
+        lm = self.landmarks[team.currentLandmark]
+        if(lm.answer!=answer):
+            team.penaltyCount+=lm.answerpenalty
+            return False
+        else:
+            if(dtn<team.clueTime):
+                team.timelog(datetime.timedelta(days=0,hours=0,minutes=0,seconds=0))
+            else:
+                team.timelog.append(dtn-team.clueTime)
+            team.currentLandmark+=1
+            team.penaltyCount+=(int(((dtn-team.clueTime)/self.timer))*lm.timePenalty)
+            if (team.penaltyCount<=lm.points):
+                team.points+=(lm.points-team.penaltyCount)
+            team.clueTime=dtn
+            team.penaltyCount=0
+            return True
+
+
+    def get_status(self, dtn, team):
+        currenttimecalc = (dtn-team.clueTime)
+        totaltime = datetime.timedelta(days=0, hours=0,minutes=0,seconds=0)
+        for t in team.timelog:
+            totaltime+=t
+        return('Points:'+str(team.points)+';You Are On Landmark:'+str(team.currentLandmark)+';Current Time:'+ str(currenttimecalc) +';Time Taken For Landmarks:'+str(totaltime))
```

Comment:
This will look much nicer in a string.format style.

Reply:
currentLandmark is a 0 based index, would it be more user friendly to convert to one based? "You are on landmark 0" vs "You are on landmark 1"

Reply:
Change the "Current Time:" text to something like "Time on current landmark question"